Sources/NIOInstrumentation/HTTPHeadersCarrier.swift

```diff
@@ -23,11 +23,12 @@ public struct HTTPHeadersExtractor: ExtractorProtocol {
     public init() {}
 
     public func extract(key: String, from headers: HTTPHeaders) -> String? {
-        return headers
+        let header = headers
             .lazy
             .filter { $0.name == key }
             .map { $0.value }
             .joined(separator: ",")
+        return header.isEmpty ? nil : header
```

Comment:
@ktoso I'm wondering whether this check should be on the array instead of the empty string. If we check on the String, users can't distinguish between no header value and an empty but present header value.

Reply:
You may be right here... on the array is more precise. In practice I wonder is empty values are meaningful at all though... is it even legal http to have an empty header, I don't remember tbh.

Reply:
Okay, even if empty header values are illegal/impossible, in my opinion, the extractor shouldn't be the "judge" of it. I'll change it to check on the array.

Reply:
Sounds good 👍